docs/high-availability.md

```diff
@@ -0,0 +1,177 @@
+# High Availability Kubernetes Clusters
+
+## Introduction
+This document describes how to build a high-available Kubernetes cluster.  This is a fairly advanced topic, 
+Users who merely want to experiment with Kubernetes are encouraged to use a configurations that are simpler to set up such as
+the simple [Docker based single node cluster instructions](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/getting-started-guides/docker.md),
+or try [Google Container Engine](https://cloud.google.com/container-engine/) for hosted Kubernetes.
+
+## Overview
+Setting up a truly reliable, highly available distributed system requires a number of distinct steps, it is akin to
+wearing underwear, pants, a belt, suspenders, another pair of underwear, and another pair of pants.  We go into each
+of these steps in detail, but a summary is given here to help guide and orient the user.
+
+The steps involved are as follows:
+   * Setting up a redundant, reliable storage layer
+   * Installing a health-checking daemon runner
+   * Turning up a clustered instance of the etcd key/value store
+   * Starting replicated, load balanced Kubernetes API servers
+   * Setting up master-elected Kubernetes scheduler and controller-manager daemons
+
+Here's what the system should look like when it's finished:
+![High availability Kubernetes diagram](high-availability/ha.png)
+
+Ready? Let's get started.
+
+## Initial set-up.
+The remainder of this guide assumes that you are setting up a 3-node clustered master, where each machine is running some flavor of Linux.
+Examples in the guide are given for Debian distributions, but they should be easily adaptable to other distributions.
+
+The easiest way to implement an HA Kubernetes cluster, is to start with an existing single-master cluster.  See [https://get.k8s.io](https://get.k8s.io)
+for easy installation instructions for single-master clusters on a variety of platforms.
+
+## Reliable storage
+The central foundation of a highly available solution is a redundant, reliable storage layer.  The number one rule of high-availability is
+to protect the data.  Whatever else happens, whatever catches on fire, if you have the data, you can rebuild.  If you lose the data, you're
+done.
+
+Clustered etcd, already replicates your storage to three nodes in your cluster.  This means that to lose data, all three nodes would need
```

Comment:
no need for comma after etcd
also, there are some assumptions baked into "Clustered etcd, already replicates your storage to three nodes in your cluster" (like, that you set up three nodes, told etcd to run on those three nodes, etc.) that aren't obvious at this point in the text. I'd suggest instead saying "Clustered etcd already replicates your storage to all of its instances. This means that to lose data, all etcd nodes would need..."